graph/resolvers/root_query.js

```diff
@@ -40,17 +40,26 @@ const RootQuery = {
     return Comments.get.load(id);
   },
 
-  async commentCount(_, {query}, {user, loaders: {Actions, Comments}}) {
+  async commentCount(_, {query}, {user, loaders: {Actions, Comments, Assets}}) {
+
     if (user == null || !user.can(SEARCH_OTHERS_COMMENTS)) {
       return null;
     }
-
-    const {action_type} = query;
+    
+    const {action_type, asset_url} = query;
 
     if (action_type) {
       query.ids = await Actions.getByTypes({action_type, item_type: 'COMMENTS'});
     }
 
+    if (asset_url) {
+      return Assets.findByUrl(asset_url)
+        .then((asset) => {
+          const newQuery = Object.assign({}, query, {asset_id: asset.id});
+          return Comments.getCountByQuery(newQuery);
+        });
+    }
+
```

Comment:
You can always do:

```js
if (asset_url) {
  let asset = await Assets.findByUrl(asset_url);
  if (asset) {
    query.asset_id = asset.id;
  }
}
```

The CommentModel does indeed support `asset_id`, so that works fine.